src/ui/public/accessibility/kbn_ui_ace_keyboard_mode.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * The `kbn-ui-ace-keyboard-mode` directive should be used on any element, that
+ * `ui-ace` is used on. It will prevent the keyboard trap, that ui-ace usually
+ * has, i.e. tabbing into the box won't give you any possibilities to leave
+ * it via keyboard again, since tab inside the textbox works like a tab character.
+ *
+ * This directive won't change anything, if the user uses the mouse. But if she
+ * tabs to the ace editor, an overlay will be shown, that you have to press Enter
+ * to enter editing mode, and that it can be left by pressing Escape again.
+ *
+ * That way the ui-ace editor won't trap keyboard focus, and won't cause that
+ * accessibility issue anymore.
+ */
+
+import angular from 'angular';
+import { uiModules } from 'ui/modules';
+import './kbn_ui_ace_keyboard_mode.less';
+import { ESC_KEY_CODE, ENTER_KEY } from 'ui_framework/services';
+
+uiModules.get('kibana').directive('kbnUiAceKeyboardMode', () => ({
+  restrict: 'A',
+  link(scope, element) {
+    const uniqueId = `uiAceKeyboardHint-${scope.$id}-${Math.floor(Math.random() * 10000)}`;
+
+    const hint = angular.element(`<div class="ui-ace-keyboard-hint" id="${uniqueId}" tabindex="0" role="application">
```

Comment:
Also, can we indent the markup to make it a little easier to read? We can also use some UI Framework components here, which will make it easier to change the font-family later.

```javascript
    const hint = angular.element(`
      <div
        class="ui-ace-keyboard-hint"
        id="${uniqueId}"
        tabindex="0"
        role="application"
      >
        <p class="kuiText kuiVerticalRhythmSmall">
          Press Enter to start editing.
        </p>

        <p class="kuiText kuiVerticalRhythmSmall">
          When you&rsquo;re done, press Escape to stop editing.
        </p>
      </div>
    `);
```

This will make it look like this:

![image](https://user-images.githubusercontent.com/1238659/29040357-8acbc9f4-7b63-11e7-9845-db136e084f18.png)